tensorboard/plugins/histogram/summary.py

```diff
@@ -0,0 +1,190 @@
+      return tf.constant([], shape=(0, 3), dtype=tf.float64)
+
+    def when_nonempty():
+      min_ = tf.reduce_min(data)
+      max_ = tf.reduce_max(data)
+      range_ = max_ - min_
+      is_singular = tf.equal(range_, 0)
+
+      def when_nonsingular():
+        bucket_width = range_ / tf.cast(bucket_count, tf.float64)
+        offsets = data - min_
+        bucket_indices = tf.cast(tf.floor(offsets / bucket_width),
+                                 dtype=tf.int32)
+        clamped_indices = tf.minimum(bucket_indices, bucket_count - 1)
+        one_hots = tf.one_hot(clamped_indices, depth=bucket_count)
+        bucket_counts = tf.cast(tf.reduce_sum(one_hots, axis=0),
+                                dtype=tf.float64)
+        edges = tf.lin_space(min_, max_, bucket_count + 1)
+        left_edges = edges[:-1]
+        right_edges = edges[1:]
+        return tf.transpose(tf.stack(
+          [left_edges, right_edges, bucket_counts]))
+
+      def when_singular():
+        center = min_
+        bucket_starts = tf.stack([center - 0.5])
+        bucket_ends = tf.stack([center + 0.5])
+        bucket_counts = tf.stack([tf.cast(tf.size(data), tf.float64)])
+        return tf.transpose(
+          tf.stack([bucket_starts, bucket_ends, bucket_counts]))
+
+      return tf.cond(is_singular, when_singular, when_nonsingular)
+
+    return tf.cond(is_empty, when_empty, when_nonempty)
+
+
+def op(name,
+       data,
+       bucket_count=None,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Create a histogram summary op.
+
+  Arguments:
+    name: A unique name for the generated summary node.
+    data: A `Tensor` of any shape. Must be compatible with `float64`.
+    bucket_count: Optional positive `int` or scalar `int32` `Tensor`.
+      (See the module-level docstring for edge cases regarding buckets
+      when there is no data or data takes only one value.)
+    display_name: Not yet supported; do not use. (Eventually: Optional
+      name for this summary in TensorBoard. Defaults to `name`.)
+    description: Not yet supported; do not use. (Eventually: Optional
+      long-form description for this summary. Markdown is supported.)
+    collections: Optional list of graph collections keys. The new
+      summary op is added to these collections. Defaults to
+      `[Graph Keys.SUMMARIES]`.
+
+  Returns:
+    A TensorFlow summary op.
+  """
+  if display_name is None:
+    display_name = name
+  summary_metadata = metadata.create_summary_metadata(
+    display_name=display_name, description=description)
+  with tf.name_scope(name):
+    tensor = _buckets(data, bucket_count=bucket_count)
+    return tf.summary.tensor_summary(name='histogram_summary',
+                                     tensor=tensor,
+                                     collections=collections,
+                                     summary_metadata=summary_metadata)
+
+
+def pb(name, data, bucket_count=None, display_name=None, description=None):
```

Comment:
We intend for `pb` to be part of the public interface, yes. There are use cases for people who use backends other than TensorFlow but still want to use TensorBoard, as well as people who use TensorFlow (e.g., perhaps your data is very large so you batch it across multiple TensorFlow runs in a loop, then manually compute some aggregate statistics and want to create a histogram displaying them—but you're not within a TensorFlow context, so rather you just call `pb` manually.

(Perhaps I'm not understanding your comment—are you saying that `pb` doesn't currently look like it's part of the public interface but should be, or the other way around?)